node/src/node/slave/spawn_handle.cpp

```diff
@@ -0,0 +1,81 @@
+#include "cocaine/detail/service/node/slave.hpp"
+
+#include "cocaine/detail/service/node/slave/machine.hpp"
+#include "cocaine/detail/service/node/slave/spawn_handle.hpp"
+#include "cocaine/detail/service/node/slave/state/handshaking.hpp"
+#include "cocaine/detail/service/node/slave/state/spawn.hpp"
+
+#include "cocaine/service/node/slave/error.hpp"
+
+#include <cocaine/logging.hpp>
+#include <cocaine/hpack/header.hpp>
+
+#include <blackhole/logger.hpp>
+
+namespace cocaine {
+namespace detail {
+namespace service {
+namespace node {
+namespace slave {
+
+spawn_handle_t::spawn_handle_t(std::weak_ptr<machine_t> _slave, std::weak_ptr<state::spawn_t> _spawning) :
+    slave(std::move(_slave)),
+    spawning(std::move(_spawning))
+{}
+
+void
+spawn_handle_t::on_terminate(const std::error_code&, const std::string& msg) {
+    //TODO: IMPLEMENT
+    assert(false);
+    throw 42;
+}
+
+void
+spawn_handle_t::on_ready() {
+    auto _spawning = spawning.lock();
+    auto _slave = slave.lock();
+    if(!_spawning || !_slave) {
+        throw cocaine::error_t("possible bug - spawning state or slave is dead");
```

Comment:
Well you require that weak pointers must be always valid. If so why do you ever use them instead of shared pointers with strong destruction orde?